lib/uklibid/include/uk/libid.h

```diff
@@ -6,4 +6,53 @@
 #ifndef LIBUKLIBID_H_
 #define LIBUKLIBID_H_
 
+#include <uk/config.h>
+#include <uk/arch/limits.h>
+#include <uk/bits/libid.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define UKLIBID_NONE __UI_MAX
+
+#define _uk_libid_static(l) \
+	__UKLIBID_ ## l
+
+/**
+ * Compile-time mapping of a library name to its unique library identifier
+ * WARNING: Any unknown library name (e.g., library not part of build) will
+ *          make this pre-processor macro fail.
+ *
+ * @param libname Non-quoted name of the library (no C-string!)
+ * @return Library identifier (unsigned int)
```

Comment:
Intuitively speaking, would a fixed type like `__u64` make more sense for identifiers?

Reply:
I was thinking also about this, but rather `__u32` or even `__u16`. Then I thought, we likely have less than `__U16_MAX` libraries in a build (16K libraries...). Then I thought, let the compiler decide an optimal default for the architecture because the values will be at most in the hundreds. A theoretical 8bit target architecture for Unikraft could still support 255 libraries. On the other hand, I agree with you too, using a fixed size integers better communicate size assumptions to calling code. The history tells us that space rockets can explode because calling code may misuse functions simply because not being aware of their original integer size assumptions. A fixed-size integer is probably also better for the case when we need to reserve space for writing libid's out somewhere... in short: you changed my mind about `unsigned int`.